worker/raft/raftclusterer/worker.go

```diff
@@ -139,12 +139,23 @@ func (w *Worker) updateConfiguration(
 ) (uint64, error) {
 	newServers := make(map[raft.ServerID]raft.ServerAddress)
 	for _, server := range details.Servers {
+		serverID := raft.ServerID(names.NewMachineTag(server.ID).String())
+		existingAddress, found := configuredServers[serverID]
+		var newAddress raft.ServerAddress
 		if server.InternalAddress == "" {
-			continue
+			if !found {
+				continue
```

Comment:
Might be worth a logger.Debug() here

Reply:
I think we decided that we don't want to inherently fall back to restoring an old address, but instead treat the address going away as indicating this machine is asking to be removed from participating in HA.